.gitlab-ci.yml

```diff
@@ -17,3 +23,50 @@ test-lisp:
   script:
     - make install-test-deps
     - make test
+
+docker-master:
+  stage: package
+  tags:
+    - dockerd
+    - github
+  only:
+    - master
+  image: docker:stable
+  script:
+    - docker build --tag rigetti/rpcq:${CI_COMMIT_SHORT_SHA} --tag rigetti/rpcq:latest .
+    - echo ${DOCKERHUB_PASSWORD} | docker login -u ${DOCKERHUB_USERNAME} --password-stdin
+    - docker push rigetti/rpcq && docker rmi -f rigetti/rpcq:latest
+
+docker-branch:
+  stage: package
+  tags:
+    - dockerd
+    - github
+  only:
+    - branches
+  except:
+    - master
+  image: docker:stable
+  script:
+    - docker build --tag rigetti/rpcq:${CI_COMMIT_REF_SLUG} .
+    - echo ${DOCKERHUB_PASSWORD} | docker login -u ${DOCKERHUB_USERNAME} --password-stdin
+    - docker push rigetti/rpcq && docker rmi -f rigetti/rpcq:${CI_COMMIT_REF_SLUG}
+
+docker-vtag:
+  stage: package
+  tags:
+    - dockerd
+    - github
+  # matches only on tags of the form vX.Y.Z
+  only:
+    - /^v(\d+\.)?(\d+\.)?(\d+)$/
+  except:
+    - branches
+  image: docker:stable
+  script:
+    - export VERSION=`cat VERSION.txt`
+    # make sure tag matches VERSION.txt
+    - if [ "${CI_COMMIT_TAG}" = "v${VERSION}" ]; then return 0; else return 1; fi
+    - docker build --tag rigetti/rpcq:$VERSION --tag rigetti/rpcq:stable --tag rigetti/rpcq:latest .
```

Comment:
also decided that we should start dropping the "v" in our docker image tags as that is the standard practice